src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -552,7 +552,7 @@ public BuildEnvironment(BuildEnvironmentMode mode, string currentMSBuildExePath,
                 MSBuildToolsDirectoryRoot = CurrentMSBuildToolsDirectory;
             }
 
-            if (mode == BuildEnvironmentMode.VisualStudio && MSBuildToolsDirectoryRoot != null)
+            if (MSBuildToolsDirectoryRoot != null)
             {
                 // Calculate potential paths to other architecture MSBuild.exe
                 var potentialAmd64FromX86 = FileUtilities.CombinePaths(MSBuildToolsDirectoryRoot, "amd64", msBuildExeName);
```

Comment:
As mentioned in Teams, `MSBuildToolsDirectoryRoot` would still be wrong due to line 538 above.

The previous logic used:
```
            // Check to see if our current folder is 'amd64'
            bool runningInAmd64 = string.Equals(currentToolsDirectory.Name, "amd64", StringComparison.OrdinalIgnoreCase);
            bool runningInARM64 = string.Equals(currentToolsDirectory.Name, "arm64", StringComparison.OrdinalIgnoreCase);
```

And effectively used that to determine whether it needed to use the parent directory or not instead of checking if it's VS and checking the current process's arch.